### PR TITLE
[18EUS] SR and OR features

### PR DESCRIPTION
**Pull request by crericha**
- Subsidies
- Grow corp to 10 share
- Buy excess shares from president
- SR pass order
- Extra home tile lay on first first operating turn
- Consent to lay another corp's home tile
- Dividend types and stock movement
- Incremental capitalization
- Train exports

### CODE REVIEW

File: lib/engine/game/g_1817/step/share_buying_with_shorts.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@ module ShareBuyingWithShorts
         # such as in 1889 for exchanging Dougo
         #
         def can_gain?(entity, bundle, exchange: false)
+          return if bundle.owner&.player?
```
Comment (outsidePasser) [inline lib/engine/game/g_1817/step/share_buying_with_shorts.rb]: Should this be in the other order like where it changed in 1822?

File: lib/engine/game/g_18_eus/game.rb [hunk 2]
```diff
@@ -153,6 +158,21 @@ class Game < Game::Base
           yellow_20|green_30|brown_40|gray_40
         ].freeze
 
+        def timeline
+          @timeline ||= [
+            'End of OR 1.1: All unsold 2 trains are exported.',
+            'End of OR 1.2: All unsold 2+ trains are exported.',
```
Comment (outsidePasser) [inline lib/engine/game/g_18_eus/game.rb:164]: Don't know if this is the right place to put it, but start of OR 2.1 should mention going to green.

File: lib/engine/game/g_18_mex/step/buy_sell_par_shares.rb [hunk 3]
```diff
@@ -17,6 +17,7 @@ def can_sell?(entity, bundle)
           end
 
           def can_gain?(entity, bundle, exchange: false)
+            return if bundle.owner&.player?
```
Comment (outsidePasser) [inline lib/engine/game/g_18_mex/step/buy_sell_par_shares.rb]: Should the !bundle check be before this line?

**Review by outsidePasser — COMMENTED** [review]
Left a couple comments in line.

File: lib/engine/game/g_1817/step/share_buying_with_shorts.rb (hunk 1 above)
Comment (crericha) [reply lib/engine/game/g_1817/step/share_buying_with_shorts.rb]: Yeah

File: lib/engine/game/g_18_eus/game.rb (hunk 2 above)
Comment (crericha) [reply lib/engine/game/g_18_eus/game.rb:164]: I lifted this timeline from USA, which has the same rule. I don't know if the timeline is only for trains or the whole game. Feel free to add to it if for the whole game.

File: lib/engine/game/g_18_mex/step/buy_sell_par_shares.rb (hunk 3 above)
Comment (crericha) [reply lib/engine/game/g_18_mex/step/buy_sell_par_shares.rb]: yes

**Pull request merged**